File: src/main/java/github/macro/build_info/gems/Slot.java
package github.macro.build_info.gems;

import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

/**
 * Created by Macro303 on 2020-Jan-22.
 */
public enum Slot {
	GREEN,
	BLUE,
	RED,
	WHITE;

	@Nullable
	public static Slot value(String name) {
		return Arrays.stream(values()).filter(tag -> tag.name().replaceAll("_", " ").equalsIgnoreCase(name)).findFirst().orElse(null);
	}
}

File: src/main/kotlin/github/macro/build_info/gems/Slot.kt
package github.macro.build_info.gems

/**
 * Created by Macro303 on 2020-Jan-17.
 */
enum class Slot {
	GREEN,
	BLUE,
	RED,
	WHITE,
	ERROR;

	companion object {
		fun value(name: String): Slot? = values().firstOrNull {
			it.name.replace("_", " ").equals(name, ignoreCase = true)
		}
	}
}